File: app/src/main/java/com/pleon/buyt/database/converter/CategoryConverter.java
package com.pleon.buyt.database.converter;

import com.pleon.buyt.model.Category;

import androidx.room.TypeConverter;

public class CategoryConverter {

    @TypeConverter
    public static Category convertToItemCategory(String name) {
        return Category.valueOf(name);
    }

    @TypeConverter
    public static String convertToItemCategoryName(Category category) {
        return category != null ? category.name() : null;
    }
}

File: app/src/main/java/com/pleon/buyt/database/converter/CategoryConverter.kt
package com.pleon.buyt.database.converter

import androidx.room.TypeConverter
import com.pleon.buyt.model.Category

class CategoryConverter {

    @TypeConverter
    fun convertToItemCategory(name: String): Category {
        return Category.valueOf(name)
    }

    @TypeConverter
    fun convertToItemCategoryName(category: Category?): String? {
        return category?.name
    }
}
